src/core/packages/chrome/browser-internal/src/ui/project/sidenav_v2/navigation/navigation.tsx

```diff
@@ -0,0 +1,405 @@
+ *   - Nodes without links that aren't panel openers are treated as section dividers and not supported in new nav - their children are flattened
+ *   - panelOpener nodes create flyout secondary navigation panels, they can't have links directly, but can have sections with links
+ * - 3rd level is used for secondary navigation (children of panelOpener):
+ *   - If all 3rd level items have links, they're treated as menu items and wrapped in a single section
+ *   - If some don't have links, they're treated as section headers with their children becoming menu items
+ * - Footer is limited to 5 items maximum (extras are dropped with warning)
+ *
+ * @param navigationTree
+ * @param navLinks
+ * @param activeNodes
+ */
+export const toNavigationItems = (
+  navigationTree: NavigationTreeDefinitionUI,
+  navLinks: Readonly<ChromeNavLink[]>,
+  activeNodes: ChromeProjectNavigationNode[][]
+): {
+  logoItem: LogoItem;
+  navItems: NavigationStructure;
+} => {
+  // HACK: extract the logo, primary and footer nodes from the navigation tree
+  let logoNode: ChromeProjectNavigationNode | null = null;
+  let primaryNodes: ChromeProjectNavigationNode[] = [];
+  let footerNodes: ChromeProjectNavigationNode[] = [];
+
+  if (navigationTree.body.length === 1) {
+    const firstNode = navigationTree.body[0];
+    if (!isRecentlyAccessedDefinition(firstNode)) {
+      logoNode = firstNode;
+      primaryNodes = firstNode.children ?? [];
+    }
+  } else {
+    warnOnce(
+      `Navigation tree has multiple root nodes. First level should have a single node for the logo and solution name.`
+    );
+  }
+  if (navigationTree.footer?.length === 1) {
+    const firstNode = navigationTree.footer[0];
+    if (!isRecentlyAccessedDefinition(firstNode)) {
+      footerNodes = firstNode.children ?? [];
+    }
+  } else {
+    warnOnce(
+      `Navigation tree footer has multiple root nodes. Footer should have a single node for the footer links.`
+    );
+  }
+
+  const logoItem: LogoItem = {
+    logoType: warnIfMissing(logoNode, 'icon', 'logoKibana') as string,
+    label: warnIfMissing(logoNode, 'title', 'Kibana'),
+  };
+
+  const toMenuItem = (navNode: ChromeProjectNavigationNode): MenuItem[] | MenuItem | null => {
+    if (!navNode) return null;
+
+    if (isRecentlyAccessedDefinition(navNode)) {
+      warnOnce(
+        `Recently accessed node "${navNode.id}" is not supported in the new navigation. Ignoring it.`
+      );
+      return null;
+    }
+
+    if (navNode.sideNavStatus === 'hidden') {
+      return null;
+    }
+
+    // Flatten accordion items into a single level with links.
+    // This is because the new navigation does not support accordion items.
+    if (navNode.renderAs === 'accordion') {
+      if (!navNode.children?.length) {
+        warnOnce(`Accordion node "${navNode.id}" has no children. Ignoring it.`);
+        return null;
+      }
+
+      const items = filterEmpty(navNode.children.flatMap(toMenuItem));
```

Comment:
Nit, for consideration: should we convert accordion nodes to `panelOpener` instead? Either way is a bit hacky. Maybe your method actually is better as it would incentivize more urgent clean-up.

Reply:
That discussion is resolved, but it still seems that flatten is closer to the end result.